src/hotspot/cpu/x86/x86.ad

```diff
@@ -10483,6 +10583,236 @@ instruct DoubleClassCheck_reg_reg_vfpclass(rRegI dst, regD src, kReg ktmp, rFlag
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType elem_bt = Matcher::vector_element_basic_type(this);
+    __ vector_saturating_op(this->ideal_Opcode(), elem_bt, $dst$$XMMRegister,
+                            $src1$$XMMRegister, $src2$$Address, false, vlen_enc);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vector_saturating_unsigned_subword_mem(vec dst, vec src1, memory src2)
+%{
+  predicate(is_subword_type(Matcher::vector_element_basic_type(n)) &&
+            n->is_SaturatingVector() && n->as_SaturatingVector()->is_unsigned());
+  match(Set dst (SaturatingAddV src1 (LoadVector src2)));
+  match(Set dst (SaturatingSubV src1 (LoadVector src2)));
+  format %{ "vector_saturating_unsigned_subword $dst, $src1, $src2" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType elem_bt = Matcher::vector_element_basic_type(this);
+    __ vector_saturating_op(this->ideal_Opcode(), elem_bt, $dst$$XMMRegister,
+                            $src1$$XMMRegister, $src2$$Address, true, vlen_enc);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vector_saturating_subword_masked_reg(vec dst, vec src, kReg mask) %{
+  predicate(is_subword_type(Matcher::vector_element_basic_type(n)) &&
+            n->is_SaturatingVector() && !n->as_SaturatingVector()->is_unsigned());
+  match(Set dst (SaturatingAddV (Binary dst src) mask));
+  match(Set dst (SaturatingSubV (Binary dst src) mask));
+  format %{ "vector_saturating_subword_masked $dst, $mask, $src" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType elem_bt = Matcher::vector_element_basic_type(this);
+    __ evmasked_saturating_op(this->ideal_Opcode(), elem_bt, $mask$$KRegister, $dst$$XMMRegister,
+                              $dst$$XMMRegister, $src$$XMMRegister, false, true, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vector_saturating_unsigned_subword_masked_reg(vec dst, vec src, kReg mask) %{
+  predicate(is_subword_type(Matcher::vector_element_basic_type(n)) &&
+            n->is_SaturatingVector() && n->as_SaturatingVector()->is_unsigned());
+  match(Set dst (SaturatingAddV (Binary dst src) mask));
+  match(Set dst (SaturatingSubV (Binary dst src) mask));
+  format %{ "vector_saturating_unsigned_subword_masked $dst, $mask, $src" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType elem_bt = Matcher::vector_element_basic_type(this);
+    __ evmasked_saturating_op(this->ideal_Opcode(), elem_bt, $mask$$KRegister, $dst$$XMMRegister,
+                              $dst$$XMMRegister, $src$$XMMRegister, true, true, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vector_saturating_subword_masked_mem(vec dst, memory src, kReg mask) %{
+  predicate(is_subword_type(Matcher::vector_element_basic_type(n)) &&
+            n->is_SaturatingVector() && !n->as_SaturatingVector()->is_unsigned());
+  match(Set dst (SaturatingAddV (Binary dst (LoadVector src)) mask));
```

Comment:
Do equivalent store operations exist we could also match for?

Reply:
ISA only supports memory operands as second source operand.